### DISCUSSION TITLE
Order by relation fields not working on one-to-many relation

### DISCUSSION THREAD

**Discussion by Umbrien** [prisma/prisma#18949] [Q&A] [answered]
Prisma Client API Reference [says ](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#orderby)I can order by relation fields:

> In [2.16.0](https://github.com/prisma/prisma/releases/2.16.0) and later, you can [order by relation fields](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#sort-post-by-the-related-user-records-name) - for example, order posts by the author's name.

But [given example](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#sort-post-by-the-related-user-records-name) will work only if I have relation where Post have only 1 Author:

```javascript
const posts = await prisma.post.findMany({
  orderBy: {
    author: {
      name: 'asc',
    },
  },
})
```

In my app, I have Product which have multiple Sizes with price in them and I want to sort by minimal product price depending on it's size:

```javascript
prisma.product.findMany({
  where: { ... },
  include: {
    sizes: {
      select: {
        price: true,
      },
      orderBy: {
        price: "asc",
      },
   },
 },
 orderBy: {
  sizes: {
    price: "asc",
  },
 },
}),
```

Hovewer I'm getting error about `orderBy sizes` type, it tells me only `_count` field is available while I need to order by price.

I guess order by relation fields is not supported right now. Is there any way to accomplish this other than raw query?

**Comment by nurul3101** [comment, marked as answer]
Hello @Umbrien 👋 

It looks like you are trying to orderBy nested relations, and I believe your use case is similar to this GitHub feature request:
- https://github.com/prisma/prisma/issues/8190

Can you have a look at this feature request?

**Reply by Umbrien** [reply]
Yes, the Problem section looks exactly like my case.

It's sad that it hasn't received any response from the prisma team for two years now.

**Reply by nurul3101** [reply]
I can understand your concerns, I would recommend adding a 👍 to the issue so that it can be prioritised. Usually, the requests that have more demand are prioritised. I'll also highlight this feature request internally with the team.
